File: rgctl-tests/ecommerce-javascript/src/coolstore/services/coolstoreProductService.js
const { createCatalogProduct } = require('../models/catalogProduct');

class CoolstoreProductService {
  constructor() {
    this.catalog = new Map();
    this.seed('329299', 'Red Fedora', 'Official Red Hat Fedora', 34.99);
    this.seed('329199', 'Forge Laptop Sticker', 'JBoss Community sticker', 8.5);
    this.seed('165613', 'Solid Performance Polo', 'Moisture-wicking polo', 17.8);
    this.seed('165614', 'Ogios T-shirt', 'CoolStore tee', 11.5);
    this.seed('165954', 'Quarkus Stickers', 'Pack of stickers', 9.99);
  }

  seed(id, name, desc, price) {
    this.catalog.set(id, createCatalogProduct(id, name, desc, price));
  }

  getProducts() {
    return Array.from(this.catalog.values());
  }

  getProductByItemId(itemId) {
    return this.catalog.get(itemId);
  }
}

module.exports = { CoolstoreProductService };

File: rgctl-tests/ecommerce-csharp/src/Ecommerce/Coolstore/Services/CoolstoreProductService.cs
using System.Collections.Concurrent;
using Ecommerce.Coolstore.Models;

namespace Ecommerce.Coolstore.Services;

public class CoolstoreProductService
{
    private readonly ConcurrentDictionary<string, CatalogProduct> _catalog = new();

    public CoolstoreProductService()
    {
        Seed("329299", "Red Fedora", "Official Red Hat Fedora", 34.99);
        Seed("329199", "Forge Laptop Sticker", "JBoss Community sticker", 8.50);
        Seed("165613", "Solid Performance Polo", "Moisture-wicking polo", 17.80);
        Seed("165614", "Ogios T-shirt", "CoolStore tee", 11.50);
        Seed("165954", "Quarkus Stickers", "Pack of stickers", 9.99);
    }

    private void Seed(string id, string name, string desc, double price)
    {
        _catalog[id] = new CatalogProduct(id, name, desc, price);
    }

    public List<CatalogProduct> GetProducts() => _catalog.Values.ToList();

    public CatalogProduct? GetProductByItemId(string itemId) =>
        _catalog.TryGetValue(itemId, out var product) ? product : null;

    public Dictionary<string, CatalogProduct> AsMap() => new(_catalog);
}
